src/module.c

```diff
@@ -8978,7 +8978,7 @@ int VM_GetClusterNodeInfo(ValkeyModuleCtx *ctx, const char *id, char *ip, char *
         return VALKEYMODULE_ERR;
     }
 
-    if (ip) valkey_strlcpy(ip, clusterNodeIp(node), NET_IP_STR_LEN);
+    if (ip) valkey_strlcpy(ip, clusterNodeIp(node, NULL), NET_IP_STR_LEN);
```

Comment:
I think we will need a new `VM_GetClusterNodeInfo` that accepts a `client *` argument next. Otherwise, modules are not ipv6 aware.

Reply:
So you think we should add `VM_GetClusterNodeInfoForClient`?

The module API doesn't expose `client *` but we could add one that takes a client ID. We have `VM_GetClientId` to get if from a ctx.

Another possibility is to let `VM_GetClusterNodeInfo` use the client in the cluster ctx by default. The "reply" APIs use the client in the ctx. I don't know if there's any limitation to that.

I suggest we postpone this for a separate PR.

Reply:
>So you think we should add VM_GetClusterNodeInfoForClient?

Yes I think we need a new API.

>I suggest we postpone this for a separate PR.

Separate PR is good but I think we should aim to get this new API in for Valkey 8.0 otherwise the dual stack support is incomplete.

Reply:
I added the module API too.

There are no tests for `VM_GetClusterNodeInfo`, so I didn't add any for `VM_GetClusterNodeInfoForClient` either.